distributed/worker.py

```diff
@@ -2488,70 +2559,69 @@ def ensure_communicating(self):
         for el in skipped_worker_in_flight:
             heapq.heappush(self.data_needed, el)
 
-    def get_task_state_for_scheduler(self, ts):
```

Comment:
This function was awkward after the refactoring. Instead, there is now a single place where the error message is generated and the task-finished is factored into a dedicated private function. Ultimately, we also should have a single place were this is generated but that's currently not possible without more work